test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/ViewExecutorTest.cs

```diff
@@ -364,7 +364,8 @@ private ViewExecutor CreateViewExecutor(DiagnosticListener diagnosticSource = nu
                 new TestHttpResponseStreamWriterFactory(),
                 new Mock<ICompositeViewEngine>(MockBehavior.Strict).Object,
                 new TempDataDictionaryFactory(new SessionStateTempDataProvider()),
-                diagnosticSource);
+                diagnosticSource,
+                new TestModelMetadataProvider());
```

Comment:
This class puts an `EmptyModelMetadataProvider` into DI. So, definitely use that class here.